### DISCUSSION TITLE
MirrorMaker2 - apply custom smt

### DISCUSSION THREAD

**Discussion by thilo-behnke** [strimzi/strimzi-kafka-operator#10285] [General]
When an event is mirrored by the MirrorMaker2 instance, I want to apply a custom smt to each event.
(This smt adds a header with custom logic to the event.)

How can I achieve this with strimzi? I can't find a configuration option to add a jar file with custom smts. Also the plugin.path is fixed, so there is no way to mount jars in an externalConfiguration mount.

**Comment by scholzj** [comment]
To add a JAR, you will need to build your own container image from a Dockerfile that uses the Strimzi image as a base image and add the JAR to the right path.

**Reply by thilo-behnke** [reply]
Thanks for the answer! I will use this approach for now. Do you think it would make sense to extend the Strimzi configuration to allow configuring plugin jars for a MirrorMaker2 instance?

In the long run the custom image approach increases the maintenance effort, i.e. when upgrading the mirror maker instances.

**Reply by scholzj** [reply]
I don't think we in general expect users to add custom plugins with the exception of Kafka Connect where you need to add connector plugins. And I have not seen this as something that comes up often for Mirror Maker 2. But if you want, feel free to open an enhancement issue. It can be triaged and used to better judge the interest etc.

> In the long run the custom image approach increases the maintenance effort, i.e. when upgrading the mirror maker instances.

Keep in mind, that someone needs to maintain the code, test it, etc. That costs a lot of effort as well. So this does not remove the effort, it just moves it on someone else. That makes sense for features that are in general demand. But not for every niche feature.

**Reply by thilo-behnke** [reply]
Sounds good, thanks!
